js/geo/transform.js

```diff
@@ -260,6 +260,16 @@ Transform.prototype = {
     },
 
     /**
+     * Like `locationPoint` but returns a raw 4d vector containing info about the transform
+     * @param {LngLat} lnglat location
+     * @returns {Array} The 4d vector pertaining to this point's transform
+     * @private
+     */
+    locationPoint3d: function(lnglat) {
+        return this.coordinatePoint3d(this.locationCoordinate(lnglat));
+    },
```

Comment:
Whoops, missed this. There are actually 2 other properties on there that are coming from a transform over pixelMatrix via Transform#_calcMatrices.

Honestly, I'm not sure what the third one is but the fourth does relate to depth.

The problem is x and y are projected into 2d space; if you were to have a true 3d point your x and y would need to be non-projected values in addition to adding z otherwise z wouldn't make sense.